packages/CopyInput/src/CopyInput.js

```diff
@@ -46,12 +48,22 @@ function CopyInput(props) {
   }
 
   return (
-    <sc.CopyInput data-pka-anchor="copy-input" {...moreProps}>
-      <Input ref={inputRef} defaultValue={value} isReadOnly={isReadOnly} {...extendedInputProps} />
+    <sc.CopyInput
+      data-pka-anchor="copy-input"
+      hasDefaultButtonBorder={hasValueContainer || !hasInputContainer}
+      {...moreProps}
+    >
+      {hasValueContainer ? <sc.Value>{value}</sc.Value> : null}
+      {!hasInputContainer || hasValueContainer ? (
+        <sc.HiddenInput ref={inputRef} defaultValue={value} />
+      ) : (
```

Comment:
Hm... 🤔 Is the hidden input _really_ needed? Maybe it's convenient and doesn't hurt, but it's probably useful any time theres no visible `<input>`, so the condition could just be `!hasInputContainer`? 

Or else the consumer could just add a redundant hidden `<input>` if their app really needs it?